platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/NonNormalizedIdentityImmutableTransformExecution.java

```diff
@@ -54,24 +53,29 @@ public ImmutableTransformExecution(
         this.fileSystemAccess = fileSystemAccess;
     }
 
+    @Override
+    protected TransformWorkspaceIdentity createIdentity(Map<String, ValueSnapshot> identityInputs, Map<String, CurrentFileCollectionFingerprint> identityFileInputs) {
+        return TransformWorkspaceIdentity.createNonNormalizedImmutable(
+            identityInputs.get(INPUT_ARTIFACT_PATH_PROPERTY_NAME),
+            identityInputs.get(INPUT_ARTIFACT_SNAPSHOT_PROPERTY_NAME),
+            identityInputs.get(SECONDARY_INPUTS_HASH_PROPERTY_NAME),
+            identityFileInputs.get(DEPENDENCIES_PROPERTY_NAME).getHash()
+        );
+    }
+
     @Override
     public void visitIdentityInputs(InputVisitor visitor) {
         super.visitIdentityInputs(visitor);
         // This is a performance hack. We could use the regular fingerprint of the input artifact, but that takes longer than
-        // capturing the normalized path and the snapshot of the raw contents, so we are using these to determine the identity
+        // capturing the normalized path and the snapshot of the raw contents, so we are using these to determine the identity.
+        // We do this because external artifact transforms typically need to identify themselves redundantly many times during a build.
+        // Once we migrate to all-scheduled transforms we won't need this optimization.
```

Comment:
❔ I don't understand why we don't need this optimization any more when we move to all-scheduled transforms. Though it is also not very important in this context.

Reply:
My understanding is that this optimization is needed because we "execute" transforms every time their output is needed, and then we de-duplicate this in the `IdentityCache` step. For external artifacts like `guava.jar` we run the identify part for every subproject where the transformed JAR is needed; for large projects the difference between using the snapshot vs using the fingerprint can add up.

Once we change this and all transforms are discovered as part of execution graph building my expectation is that we can do this de-duplication based on the identity of the transformed dependency instead. This is just an assumption, though, that I haven't verified, so I'm okay with removing the comment here.

Reply:
I understand that we could do the de-duplication while scheduling, though that also means we'd do the identification there. In other words for de-duplication we need to identify a transform, and if my understanding is correct that doesn't change whether we schedule the transforms or not.

Reply:
As discussed in person: my expectation is that we can de-duplicate while creating the execution graph by observing the input artifact's coordinates, and that'll be somehow cheaper. Or we can use the input snapshot there if that doesn't work. But here we will be able to use the full fingerprint instead.

I've removed the comment in any case.